src/ipa_pc/data_structures.rs

```diff
@@ -27,12 +29,55 @@ pub struct UniversalParams<G: AffineCurve> {
     pub hash: Vec<u8>
 }
 
-impl<G: AffineCurve> PCUniversalParams for UniversalParams<G> {
+impl<G: AffineCurve> PCParameters for Parameters<G> {
+    type CommitterKey = CommitterKey<G>;
+    type VerifierKey = VerifierKey<G>;
+    type Error = Error;
+
     fn max_degree(&self) -> usize {
         self.comm_key.len() - 1
     }
+
     fn get_hash(&self) -> &[u8] { self.hash.as_slice() }
-    fn copy_params(&mut self, other: &Self) {
+
+    /// Trims the base point vector of the setup function to a custom segment size
+    fn trim(
+        &self,
+        // the segment size (TODO: let's rename it!)
+        supported_degree: usize,
+    ) -> Result<(Self::CommitterKey, Self::VerifierKey), Self::Error> {
```

Comment:
why not output a ``Parameters``?

Reply:
Parameters struct doesn't contain commiter / verifier key. Do you want to put them inside parameters in some way?